generators/server/templates/src/main/java/package/gateway/accesscontrol/_AccessControlFilter.java

```diff
@@ -0,0 +1,92 @@
+package <%=packageName%>.gateway.accesscontrol;
+
+import <%=packageName%>.config.JHipsterProperties;
+import com.netflix.zuul.ZuulFilter;
+import com.netflix.zuul.context.RequestContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cloud.netflix.zuul.filters.ProxyRouteLocator;
+import org.springframework.http.HttpStatus;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Zuul filter for restricting access to backend micro-services endpoints
+ */
+public class AccessControlFilter extends ZuulFilter{
+
+    private final Logger log = LoggerFactory.getLogger(AccessControlFilter.class);
+
+    @Inject
+    private ProxyRouteLocator routeLocator;
+
+    @Inject
+    private JHipsterProperties jHipsterProperties;
+
+    @Override
+    public String filterType() {
+        return "pre";
+    }
+
+    @Override
+    public int filterOrder() {
+        return 0;
+    }
+
+    @Override
+    // Filter requests on endpoints that are not in the list of authorized microservices endpoints
+    public boolean shouldFilter() {
+        Map<String, String> routes = routeLocator.getRoutes();
+        String requestUri = RequestContext.getCurrentContext().getRequest().getRequestURI();
+
+        // If the request Uri does not start with the path of the authorized endpoints, we block the request
+        for (Map.Entry<String,String> route : routes.entrySet()) {
+            String serviceUrl = route.getKey();
+            String serviceName = route.getValue();
+
+            // If this route correspond to the current request URI
+            if(requestUri.startsWith(serviceUrl.substring(0,serviceUrl.length() - 2))){
+                if (isAuthorizedRequest(serviceUrl,serviceName,requestUri)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    private boolean isAuthorizedRequest(String serviceUrl, String serviceName, String requestUri) {
+        Map<String, List<String>> authorizedMicroservicesEndpoints = jHipsterProperties.getGateway().getAuthorizedMicroservicesEndpoints();
+
+        // If the authorized endpoints list was left empty for this route, all access are allowed
+        if(authorizedMicroservicesEndpoints.get(serviceName) == null){
+            log.debug("Access Control: allowing access for: " + requestUri + ", as no access control policy has been set up for service: " + serviceName);
```

Comment:
For the log.debug you need to use the `log.debug("my text {}", value);` pattern as it's more efficient